src/main/groovy/edu/oregonstate/mist/musicapi/resources/AlbumResource.groovy

```diff
@@ -0,0 +1,268 @@
+package edu.oregonstate.mist.musicapi.resources
+
+import edu.oregonstate.mist.api.AuthenticatedUser
+import edu.oregonstate.mist.api.Resource
+import edu.oregonstate.mist.musicapi.core.Album
+import io.dropwizard.auth.Auth
+import java.util.regex.Pattern
+import javax.ws.rs.GET
+import javax.ws.rs.POST
+import javax.ws.rs.PUT
+import javax.ws.rs.DELETE
+import javax.ws.rs.Path
+import javax.ws.rs.PathParam
+import javax.ws.rs.Produces
+import javax.ws.rs.Consumes
+import javax.ws.rs.core.Response
+import javax.ws.rs.core.Response.ResponseBuilder
+import org.skife.jdbi.v2.DBI
+import org.skife.jdbi.v2.Handle
+import org.skife.jdbi.v2.util.IntegerMapper
+
+@Path('/album')
+@groovy.transform.TypeChecked
+class AlbumResource extends Resource {
+    private DBI dbi
+    private Pattern releaseDatePattern = Pattern.compile(/^[0-9]{4}-[0-9]{2}-[0-9]{2}$/) // YYYY-MM-DD
+
+    public AlbumResource(DBI dbi) {
+        this.dbi = dbi
+    }
+
+    /**
+     * Responds to GET requests by returning an album
+     *
+     * @return album
+     */
+    @GET
+    @Path('{id}')
+    @Produces("application/json")
+    Response getAlbum(@Auth AuthenticatedUser authenticatedUser, @PathParam("id") int id) {
+        def h = this.dbi.open()
+        try {
+            def result = this.getAlbumById(h, id)
+            if (result == null) {
+                return this.notFound().build()
+            }
+            return this.ok(result).build()
+        } finally {
+            h.close()
+        }
+    }
+
+    /**
+     * Creates a new album
+     */
+    @POST
+    @Consumes("application/json")
+    @Produces("application/json")
+    Response createAlbum(@Auth AuthenticatedUser _, Album album) {
+        def h = this.dbi.open()
+        try {
+            // Get artist id
+            def artistId = this.getOrCreateArtist(h, album.artist)
+
+            // Get status id
+            def statusId = this.getStatus(h, album.status)
+            if (statusId == null) {
+                return this.badRequest('invalid status').build()
```

Comment:
Yes. Artists aren't cleaned up when albums are deleted, either.
I guess there's no harm in changing the order here, though, even if it wouldn't completely solve the problem.